### DISCUSSION TITLE
Tobias Koppers - Webpack, Turbopack

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#133] [Announcements]
## Background

- https://turbo.build/pack
- https://webpack.js.org
- https://twitter.com/wsokra
- https://github.com/sokra

## Themes

- webpack/turbopack
- monorepos

## Intro

Andrew: Hello, welcome to the devtools.fm podcast. This is a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin.

Justin: Hey everyone! Our guest today is the one and only Tobias Koppers. Tobias is the creator of webpack (and now turbopack). Tobias, we're really excited to have you and and to chat about what you've been up to at Vercel. Before that though, could you tell our listeners a bit more about yourself?

### Webpack

- [x] Why did you first build webpack?
- [x] Have you been surprised by the success? Has it been hard to steer that beast?
- [x] What an example of community creativity that's inspired you? (I'm fond of module federation)

### Turbopack

- [x] Why build a new tool? What baggage from webpack did you think needed to be shed?
- [x] How related to turborepo is the project? Is it more than just the name?
- [x] Are there any plans to build other tools on top of the turbo core?
- [x] A large part of webpack is the ability to control what happens with loaders and plugins, will turbo pack have similar power through plugins?
- [x] What language will plugins be authored in? Are there tradeoffs between js vs native?
- [x] Does turbo pack rely on esmodules more? Or does it chunk code like webpack?
- [ ] ~What are your personal goals with tuebopack? What things do you want to make better?~
- [x] Are there assumptions you can make now that improve bundling and tree shaking?

### Future

- [x] What's the future of webpack maintenance?
- [ ] When do you think turbo pack will be production ready?
- [ ] What do you think the future of building for the web looks like?

**Comment by just-be-dev** [comment]
Tooltips

- https://github.com/julusian/node-elgato-stream-deck
- https://progrium.com/blog/

**Comment by hipstersmoothie** [comment]
https://maath.pmnd.rs
https://github.com/vercel/satori

**Comment by sokra** [comment]
* https://github.com/salsa-rs/salsa
* https://v8.dev/blog
